src/solvers/macros.rs

```diff
@@ -0,0 +1,56 @@
+#[allow(unused_macros)]
+macro_rules! mipgap_tests {
```

Comment:
Why does this need to be a macro ? Can't this be a generic function ?

See https://github.com/rust-or/good_lp/blob/main/tests/static_solver.rs#L5

Reply:
The macro generates several tests, each in a separate function.  Would you move all tests into a single generic function with multiple assertions, or would each test be a separate generic function?

Reply:
I think a single large generic function is with, and then multiple tiny test functions that just call the large function is fine.

```rust
fn mip_gap_assertions<S: Solver>(solver: S) {

}

#[test]
fn mip_gap_solver1() { mip_gap_assertions(solver1) }
```

I think moving some of the tests to doctests would be useful too : less verbose, and would serve as documentation too.

See for instance https://github.com/rust-or/good_lp/blob/c500648bd2cdf414f409c16d2d739213a5ce0a2f/src/solvers/mod.rs#L206-L226